# This applies various fixes to the systel.edf.cfg config file
# (include, link and compiler options) to make EasyBuilding easier
# (done here for foss- or gfbf-based builds)
#
# Author: Wouter Lampaert
--- telemac-mascaret-v8p5/configs/systel.edf.cfg.orig   2024-01-22 15:07:57.102808764 +0100
+++ telemac-mascaret-v8p5/configs/systel.edf.cfg        2024-01-22 15:09:06.528064815 +0100
@@ -2,7 +2,8 @@
 # ____/ TELEMAC Project Definitions /______________________________/
 #
 [Configurations]
-configs: S10.gfortran.dyn S10.gfortran.debug
+configs: easybuild easybuild.debug
+         S10.gfortran.dyn S10.gfortran.debug
          S10.nag.dyn S10.nag.debug
          S11.gfortran.dyn S11.gfortran.debug
          S11.nag.dyn S11.nag.debug
@@ -44,21 +45,25 @@
 #
 # METIS info
 #
+incs_metis: -I$METISHOME/include
 libs_metis: -L$METISHOME/lib -lmetis
 #
 # MED info
 #
-incs_med: -I$MEDHOME/include
-flags_med: -DHAVE_MED
-libs_med: -lm -L$MEDHOME/lib -lmed
+incs_med:
+flags_med:
+libs_med:
 #
 # MUMPS info
 #
-incs_mumps: -I$MUMPSHOME/include
+incs_mumps: -I/apps/leuven/rocky8/icelake/2022b/software/MUMPS/5.6.1-foss-2022b-metis/include
 flags_mumps: -DHAVE_MUMPS
-libs_mumps: -L$MUMPSHOME/lib -ldmumps -lmumps_common -lpord
-            -L$SCALAPACKHOME/lib -lscalapack
-            -lblas
+libs_mumps: -L$MUMPSHOME/lib/libdmumps.a
+            -L$MUMPSHOME/lib/libmumps_common.a
+            -L$MUMPSHOME/lib/libpord.a
+            -L$SCALAPACKHOME/lib/libscalapack.so
+            -L$FLEXIBLASHOME -ldmumps -lmumps_common -lpord
+            -lflexiblas -lgfortran -lpthread -lm -ldl
 libs_mumps_mkl: -L$MUMPSHOME/lib -ldmumps -lmumps_common -lpord
                 -lmkl_scalapack_lp64 -lmkl_blacs_intelmpi_lp64
                 -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -lmkl_def
@@ -66,19 +71,24 @@
 #
 # AED info
 #
-incs_aed: -I$AEDHOME/include
-flags_aed: -DHAVE_AED2
-libs_aed: -L$AEDHOME/lib -laed2
+incs_aed:
+flags_aed:
+libs_aed:
 #
 # GOTM info
 #
-incs_gotm: -I $GOTMHOME/include
-flags_gotm: -DHAVE_GOTM
-libs_gotm: -L$GOTMHOME/lib -lturbulence -lutil
+incs_gotm:
+flags_gotm:
+libs_gotm:
+#
+# NETCDF info
+#
+incs_netcdf: -I$NETCDFHOME/include -I$NETCDFMINFORTRANHOME/include
+libs_netcdf: -L$NETCDFHOME/lib -lnetcdf -L$NETCDFMINFORTRANHOME/lib -lnetcdff
 #
 # GFortran info
 #
-fflags_gfo: -cpp -fconvert=big-endian -frecord-marker=4 -DHAVE_I16
+fflags_gfo: -fallow-invalid-boz -cpp -fconvert=big-endian -frecord-marker=4 -DHAVE_I16
 fflags_debug_gfo: -g -Wall -fcheck=all -fbacktrace -fbounds-check -finit-integer=-1 -finit-real=nan -ffpe-trap=invalid,zero,overflow
 #
 # Intel Fortran compilation flags
@@ -124,12 +134,50 @@
 cmd_lib:    [ar] [lib_flags] <libname> <objs>
 cmd_exe:    [fc] [exe_flags] -o <exename> <objs> <libs>
 #
-incs_all: [incs_med] [incs_mumps] [incs_aed] [incs_gotm]
-libs_all: [libs_mumps] [libs_med] [libs_metis] [libs_aed] [libs_gotm]
+incs_all: [incs_med] [incs_mumps] [incs_aed] [incs_gotm] [incs_netcdf] [incs_metis]
+libs_all: [libs_mumps] [libs_med] [libs_metis] [libs_aed] [libs_gotm] [libs_netcdf]
 #
 cmd_obj_c: gcc -c [cflags] <srcName> -o <objName>
 cflags:
 #
+# _____            __________________________________
+# ____/ Easybuild /_________________________________/
+#
+# GFortran
+#
+[easybuild]
+brief: Dynamic build using GFortran and OpenMPI. Config used to build with Easybuild.
+options: api
+#
+f2py_name: f2py
+pyd_fcompiler: gnu95
+sfx_lib:    .so
+#
+obj_flags: -O2 -ftree-vectorize -march=native -fno-math-errno -fPIC [fflags_gfo] [flags_mpi] [flags_mumps] [flags_med] [flags_aed] [flags_gotm]
+lib_flags: -fPIC -shared [fflags_gfo]
+#
+exe_flags: -fPIC [fflags_gfo]
+#
+cmd_lib:    [fc] [lib_flags] -o <libname> <objs>
+cflags: -fPIC
+#
+[easybuild.debug]
+brief:  Dynamic build using GFortran and OpenMPI. Config used to build with Easybuild. Debug mode.
+options: api
+#
+f2py_name: f2py
+pyd_fcompiler: gnu95
+sfx_lib:    .so
+#
+obj_flags: -c -O0 -fPIC [fflags_gfo] [fflags_debug_gfo] [flags_mpi] [flags_mumps] [flags_med] [flags_aed] [flags_gotm]
+lib_flags: -fPIC -shared [fflags_gfo] [fflags_debug_gfo]
+#
+exe_flags: -fPIC [fflags_gfo]
+#
+cflags: -fPIC
+#
+cmd_lib: [fc] [lib_flags] -o <libname> <objs>
+#
 # _____             ________________________________________________
 # ____/ Scibian 10 /_______________________________________________/
 #
